HIP/hip-904-frictionless-airdrop.md

```diff
@@ -8,232 +8,422 @@ type: Standards Track
-will continue to have a used_auto_associations default of 0. Accounts created before this feature is released will remain unchanged.
+- The account holder no longer prepays for unused slots regardless of the setting
+  for `max_auto_associations`. Slots are only paid for when used, and are initially paid for by the
+  sender during automatic associations. The account is responsible for covering rent for each
+  association after the first auto-renewal period. `used_auto_associations` is incremented every
+  time an auto-association is made.
+- The default value for used_auto_associations for new automatically-created accounts will now be
+  -1. This means, if an account is created automatically (by performing a token transfer to an alias
+  that does not yet exist, see [HIP-583](https://hips.hedera.com/hip/hip-583)), then the automatically created account is configured
+  with unlimited automatic token associations. Accounts created by using the
+  [AccountCreateTransaction](https://docs.hedera.com/hedera/sdks-and-apis/sdks/accounts-and-hbar/create-an-account) (HAPI: [CryptoCreate](https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto))
+  will continue to have a used_auto_associations default of 0. Accounts created before this feature
+  is released will remain unchanged.
 
 ### Crypto Transfers
 
-Normal crypto-transfers remain the same as today, with one difference. If Account A sends a token to Account B, and Account B is not associated with that token but has an available auto-association slot, then Account A will pay for that association and the first auto-renewal period’s rent, in addition to the typical transfer costs. In summary previously the reciever would pay for an association in advance, not the payer of the transaction.
+Normal crypto-transfers remain the same as today, with one difference. If Account A sends a token to
+Account B, and Account B is not associated with that token but has an available auto-association
+slot, then Account A will pay for that association and the first auto-renewal period’s rent, in
+addition to the typical transfer costs. In summary previously the reciever would pay for an
+association in advance, not the payer of the transaction.
 
 ![Crypto Transfer Transaction](../assets/hip-904/frictionless-airdrop-cryptoTransferTransaction.png)
 
 ### Airdrops
 
-Airdrops are similar to crypto transfers, but differ in one fundamental way - when an airdrop is performed and the receiver does not have available or automatic association slots, rather than failing, the transfer will be kept in state as a *pending transfer*. To represent this new semantic, there will be a new `TokenAirdropTransaction` meant for use specifically by those attempting an airdrop, with the following semantics:
+Airdrops are similar to crypto transfers, but differ in one fundamental way - when an airdrop is
+performed and the receiver does not have available or automatic association slots, rather than
```

Comment:
Will a regular crypto transfer fail in the same scenario of a receiver not having available or automatic associations?

Reply:
Yes, a regular transfer to an account that is not already associated to a token, and the account does not have available automatic association slots, the transfer will fail.